### PR TITLE
Introduce Gradle Managed Devices

### LINKED ISSUE #783: GardenActivityTest can't be run
Even after #764 is fixed, GardenActivityTest can't be run due to the following error.

Looks like the WorkManager isn't properly initialized in the test.

```
➜  sunflower git:(fix_instrumented_deps) ✗ ./gradlew connectedCheck

> Configure project :app
WARNING: The Gradle Managed Device DSL and associated tests are experimental

> Task :app:connectedDebugAndroidTest
Starting 15 tests on Pixel 4 - 12

com.google.samples.apps.sunflower.GardenActivityTest > clickAddPlant_OpensPlantList[Pixel 4 - 12] FAILED 
        java.lang.IllegalStateException: WorkManager is not initialized properly.  You have explicitly disabled WorkManagerInitializer in your manifest, have not manually called WorkManager#initialize at this point, and your Application does not implement Configuration.Provider.
        at androidx.work.impl.WorkManagerImpl.getInstance(WorkManagerImpl.java:158)
Tests on Pixel 4 - 12 failed: There was 1 failure(s).

```

### PR DESCRIPTION
**Pull request by thagikura**
Introduce Gradle Managed Devices 
Also this PR enables instrumented tests as part of the GitHub Actions    
Also this PR fixes #783

### CODE REVIEW

**Comment by thagikura** [comment]
Is it possible to have a look? @yuuki3655 thanks!

**Comment by arriolac** [comment]
Please rebase/merge `main` into your branch. #787 converted `build.gradle` to `build.gradle.kts` file so you'll need to convert some of the additions you made to Kotlin script. Will let @yuuki3655 give the LGTM for this PR. Thanks @thagikura

**Comment by thagikura** [comment]
@arriolac ok, rebased main and added the GMD definition in `build.gradle.kts`.

File: app/build.gradle.kts [hunk 1]
```diff
@@ -86,6 +86,19 @@ android {
...
   }
+
+  testOptions {
+    managedDevices {
+      devices {
+        maybeCreate<com.android.build.api.dsl.ManagedVirtualDevice>("pixel2api27").apply {
+          device = "Pixel 2"
+          apiLevel = 27
+          systemImageSource = "aosp"
+          require64Bit = false
```
Comment (yuuki3655) [inline app/build.gradle.kts:97]: This is optional as the default value is false.

**Review by yuuki3655 — COMMENTED** [review]
Gradle Managed Device config looks good to me.

File: app/build.gradle.kts (hunk 1 above)
Comment (thagikura) [reply app/build.gradle.kts]: Thanks. Removed this property.

**Comment by thagikura** [comment]
Thanks @yuuki3655 ! @arriolac is it possible to have a look from the sunflower point of view.

**Review by arriolac — APPROVED** [review]

**Pull request merged**